app/controllers/orders_controller.rb

```diff
@@ -153,17 +153,16 @@ def remove
   # POST /orders/:id/choose_account
   def choose_account
     if request.post?
-      begin
-        account = Account.find(params[:account_id])
-        raise ActiveRecord::RecordNotFound unless account.can_be_used_by?(@order.user)
-      rescue
-      end
+      account = Account.find(params[:account_id])
       if account
         success = true
         @order.transaction do
           begin
             @order.invalidate
-            @order.update_attributes!(:account => account)
+            @order.update_attributes!(account_id: account.id)
+            @order.order_details.each do |order_detail|
```

Comment:
I believe this is redundant since the `after_save` hook on `Order` should take care of this (and it keeps it in the transaction).